vmm_tests/vmm_tests/tests/tests/x86_64/openhcl_linux_direct.rs

```diff
@@ -396,6 +396,90 @@ async fn openhcl_linux_storvsp_dvd(config: PetriVmConfigOpenVmm) -> Result<(), a
+
+    let vtl2_lun = 5;
+    let scsi_instance = Guid::new_random();
+
+    let mut vtl2_settings = None;
+
+    let (vm, agent) = config
+        .with_vmbus_redirect()
+        .with_custom_config(|c| {
+            c.vpci_devices.extend([VpciDeviceConfig {
+                vtl: DeviceVtl::Vtl2,
+                instance_id: NVME_INSTANCE,
+                resource: NvmeControllerHandle {
+                    subsystem_id: NVME_INSTANCE,
+                    max_io_queues: 64,
+                    msix_count: 64,
+                    namespaces: vec![NamespaceDefinition {
+                        nsid: vtl2_nsid,
+                        disk: (LayeredDiskHandle::single_layer(RamDiskLayerHandle {
+                            len: Some(nvme_disk_sectors * sector_size),
+                        })
+                        .into_resource()),
+                        read_only: false,
+                    }],
+                }
+                .into_resource(),
+            }]);
+        })
+        .with_custom_vtl2_settings(|v| {
+            v.dynamic.as_mut().unwrap().storage_controllers.push(
+                vtl2_settings_proto::StorageController {
+                    instance_id: scsi_instance.to_string(),
+                    protocol: vtl2_settings_proto::storage_controller::StorageProtocol::Scsi.into(),
+                    luns: vec![vtl2_settings_proto::Lun {
+                        location: vtl2_lun,
+                        device_id: Guid::new_random().to_string(),
+                        vendor_id: "OpenVMM".to_string(),
+                        product_id: "Disk".to_string(),
+                        product_revision_level: "1.0".to_string(),
+                        serial_number: "0".to_string(),
+                        model_number: "1".to_string(),
+                        is_dvd: true,
+                        physical_devices: Some(vtl2_settings_proto::PhysicalDevices {
+                            r#type: vtl2_settings_proto::physical_devices::BackingType::Single
+                                .into(),
+                            device: Some(vtl2_settings_proto::PhysicalDevice {
+                                device_type: vtl2_settings_proto::physical_device::DeviceType::Nvme
+                                    .into(),
+                                device_path: NVME_INSTANCE.to_string(),
+                                sub_device_path: vtl2_nsid,
+                            }),
+                            devices: vec![],
+                        }),
+                        ..Default::default()
+                    }],
+                    io_queue_depth: None,
+                },
+            );
+            vtl2_settings = Some(v.clone());
+        })
+        .run()
+        .await?;
+
+    let read_drive = || agent.read_file("/dev/sr0");
+    let b = read_drive().await.context("failed to read dvd drive")?;
+    let len = nvme_disk_sectors * sector_size;
+    if b.len() as u64 != len {
```

Comment:
would it be useful to check more than just the length of the returned data?

i.e: add a new `RamDiskWithDataLayerHandle`, that allows the disk to be pre-populated with a `Vec<u8>` of data that you can then compare against?

Reply:
Sure would. Do you have any objection to adding that as a follow-up PR?

Reply:
I agree in principle. However, I am struggling a bit with the logic underneath the RamDiskLayerHandle in the VMM tests. Where does the actual ramdisk get allocated?

Reply:
Yeah, this ties into some of the resource/resolver infrastructure in OpenVMM, which can be a bit tricky to wrap your head around. Once you know where to look, it's not too bad, but the fact you can't simply "jump to definition" your way between the Handle to the underlying call to `RamDisk::new` is a bit unfortunate.

Start by taking a look at `/vm/devices/storage/disklayer_ram/src/resolver.rs`. I think you'll find it shouldn't be _too_ hard to quickly add this new handle type.

That said - if you find yourself falling down a rabbit here, I wouldn't block adding this baseline collateral on it.

Reply:
I have been looking into this for a bit now. It looks like the SQLite backend is able to make this work because its handle just contains a string with the disk path of the database. However, if we put the actual RamDiskLayer here, it would be a looping reference, and I don't really think it's a good idea to provide a direct reference to the RamState or the RwLock surrounding it. Any thoughts on the design here?

Reply:
I wouldn't be opposed to this... but a much simpler alternative might be to create a temporary raw file with `tempfile::tempfile()`, write whatever you want to it, and send it over as a `FileDisk`.

Reply:
In fact, I think I'd prefer that. In the future, we could also consider a section-backed file disk, so that we can use a memory/pagefile-backed file. But let's not extend the existing RAM disk to support external data.

Reply:
I typically bias away from tempfile based solutions on purely aesthetic grounds, but it's not unreasonable for something like this. Plus, thinking ahead - it like this could be a useful test to run with the Hyper-V petri backend, which would probably require using a concrete backing file, right?

Reply:
@eric135 , please file a follow-up issue to track this conversation before closing out this PR.

Reply:
I believe this issue has been satisfactorily resolved with the use of tempfile.